pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -215,14 +221,53 @@ func (i *KumaInjector) proxyConcurrencyFor(pod *kube_core.Pod) (int64, error) {
 	return ncpu, nil
 }
 
-func (i *KumaInjector) NewSidecarContainer(pod *kube_core.Pod, ns *kube_core.Namespace) (kube_core.Container, error) {
-	mesh := meshName(pod, ns)
-	env, err := i.sidecarEnvVars(mesh, pod.GetAnnotations())
+func (i *KumaInjector) NewSidecarContainer(
+	pod *kube_core.Pod,
+	ns *kube_core.Namespace,
+) (kube_core.Container, error) {
+	container, err := i.newDpContainer(pod.Annotations, ns)
+	if err != nil {
+		return container, err
+	}
+
+	// On versions of Kubernetes prior to v1.15.0
+	// ServiceAccount admission plugin is called only once, prior to any mutating web hook.
+	// That's why it is a responsibility of every mutating web hook to copy
+	// ServiceAccount volume mount into containers it creates.
+	container.VolumeMounts = i.NewVolumeMounts(&pod.Spec)
+
+	container.Name = util.KumaSidecarContainerName
+
+	return container, nil
+}
+
+func (i *KumaInjector) NewGatewayContainer(
```

Comment:
I had that before but I changed it to this, the idea being the injector is the only thing that needs to know the particulars of the Kuma container.

Reply:
The way I'm thinking about is that the concern of the injector is injecting sidecars. It shouldn't be concerned with gateways (or zone ingress imho) because that's not a pod that needs a sidecar injected. It does seem reasonable to have some common code for deploying kuma-dp pods that aren't sidecars.

Reply:
Resolving this, tracked as a TODO in the PR